nntrainer/tensor/manager.h

```diff
@@ -173,8 +173,6 @@ class Manager {
    */
   void reset() {
     deallocateTensors(true);
-    weights_allocated = false;
```

Comment:
I reckon deallocateTensors changes weights_allocated to false?

Reply:
deallocateTensors() calls deallocateWeights() which sets weights_allocated to false.